scripts/angry-tock.js

```diff
@@ -0,0 +1,286 @@
+const holidays = require('@18f/us-federal-holidays');
+const moment = require('moment-timezone');
+const scheduler = require('node-schedule');
+
+const TOCK_API_URL = process.env.HUBOT_TOCK_API;
+const TOCK_TOKEN = process.env.HUBOT_TOCK_TOKEN;
+
+const ANGRY_TOCK_TIMEZONE =
+  process.env.ANGRY_TOCK_TIMEZONE || 'America/New_York';
+const ANGRY_TOCK_FIRST_ALERT = moment(
+  process.env.ANGRY_TOCK_FIRST_TIME || '10:00',
+  'HH:mm'
+);
+const ANGRY_TOCK_SECOND_ALERT = moment(
+  process.env.ANGRY_TOCK_SECOND_TIME || '16:00',
+  'HH:mm'
+);
+
+/**
+ * Join two arrays on a given property. Similar to a SQL JOIN.
+ * @param {Array} arrayOne first array to be merged
+ */
+const join = arrayOne => ({
+  /**
+   * @param {Array} arrayTwo second array to be merged
+   */
+  with: arrayTwo => ({
+    /**
+     * @param {string} key property to join the arrays on
+     */
+    on: key =>
+      arrayOne
+        .filter(a => arrayTwo.some(b => b[key] === a[key]))
+        .map(a => ({
+          ...a,
+          ...(arrayTwo.find(b => b[key] === a[key]) || {})
+        }))
+  })
+});
+
+/**
+ * Get the current time in the configured timezone.
+ * @returns {Moment} A moment object representing the current time in the
+ *   configured timezone.
+ */
+const m = () => moment.tz(ANGRY_TOCK_TIMEZONE);
+
+/**
+ * Fetch a list of Slack users in the workspace that this bot is in.
+ * @async
+ * @param {Object} robot Hubot robot object
+ * @returns {Promise<Array<Object>>} A list of Slack users.
+ */
+const getSlackUsers = async robot =>
+  new Promise((resolve, reject) => {
+    robot.adapter.client.web.users.list((err, response) => {
+      if (err) {
+        return reject(new Error(err));
+      }
+      return resolve(response.members);
+    });
+  });
```

Comment:
You can leverage [`util.promisify`](https://nodejs.org/dist/latest/docs/api/util.html#util_util_promisify_original) if you’re interested — 

```javascript
const { promisify } = require('util');

const getSlackUsers = async robot => {
  const response = await promisify(robot.adapter.client.web.users.list)();
  return response.members;
};
```

… or perhaps too far: 

```javascript
const { promisify } = require('util');

const getSlackUsers = async robot => 
  (await promisify(robot.adapter.client.web.users.list)()).members;
```

Reply:
(the same technique could be repeated below as well, but won’t repeat myself)

Reply:
TIL Node has a built-in promisify! I'm going to switch over to that for maintainability. Thanks! 😄